app/code/community/Nexcessnet/Turpentine/etc/config.xml

```diff
@@ -441,46 +441,22 @@
             </updates>
         </layout>
         <events>
-            <controller_action_predispatch>
-                <observers>
-                    <controller_action_before>
-                        <class>turpentine/observer_esi</class>
-                        <method>hookToControllerActionPreDispatch</method>
-                    </controller_action_before>
-                </observers>
-            </controller_action_predispatch>
-            <!--controller_action_postdispatch>
-                <observers>
-                    <controller_action_after>
-                        <class>turpentine/observer_esi</class>
-                        <method>hookToControllerActionPostDispatch</method>
-                    </controller_action_after>
-                </observers>
-            </controller_action_postdispatch-->
-            <add_to_cart_before>
+            <controller_action_predispatch_checkout_cart_add>
```

Comment:
If you look at Mage_Core_Controller_Varien_Action::preDispatch(), you'll see that there are 3 `Mage::dispatchEvent()` at the end of the method. One is `controller_action_predispatch` straight up, which is what was previously being observed here. That means our various observers were running on EVERY action on the Magento frontend. Instead we can observe the more specific `'controller_action_predispatch_' . $this->getFullActionName()` and eliminate most of that overhead.